src/librustc_passes/error_codes.rs

```diff
@@ -286,6 +286,34 @@ fn main() {
 ```
 "##,
 
+E0567: r##"
+Generics have been used on an auto trait.
+
+Erroneous code example:
+
+```compile_fail,E0567
+#![feature(optin_builtin_traits)]
+
+auto trait Generic<T> {} // error!
+
+fn main() {}
+```
+
+Since an auto trait is implemented on all existing types, it cannot have
+generics for the simple reason that there is no way for the compiler to guess
+what would be those generics.
```

Comment:
> Since an auto trait is implemented on all existing types, the 
compiler would not be able to infer the types of the trait's generic 
parameters.

I think this might be a slightly clearer wording here?